pkg/util/util.go

```diff
@@ -53,104 +53,104 @@ func ParseEndpoint(endpoint string) (string, string, error) {
+	for _, value := range slice {
+		if value == element {
+			return true
 		}
 	}
+	return false
+}
 
-	slices = append(slices, input[lastComma:])
+func EncodeDeletionTag(input string) string {
+	input = strings.ReplaceAll(input, "\"", " @ ")
+	input = strings.ReplaceAll(input, ",", " . ")
+	input = strings.ReplaceAll(input, "[", " - ")
+	input = strings.ReplaceAll(input, "]", " _ ")
+	input = strings.ReplaceAll(input, "{", " + ")
+	input = strings.ReplaceAll(input, "}", " = ")
+	return input
+}
 
-	var values []string
-	for _, val := range slices {
-		if strings.HasPrefix(val, "{") {
-			val = strings.Trim(val, "{}")
-		}
-		values = append(values, val)
-	}
-	return values
+func DecodeDeletionTag(input string) string {
+	input = strings.ReplaceAll(input, " @ ", "\"")
+	input = strings.ReplaceAll(input, " . ", ",")
+	input = strings.ReplaceAll(input, " - ", "[")
+	input = strings.ReplaceAll(input, " _ ", "]")
+	input = strings.ReplaceAll(input, " + ", "{")
+	input = strings.ReplaceAll(input, " = ", "}")
+	return input
 }
 
-func MapValues(input []string) (map[string]*string, error) {
-	valueMap := make(map[string]*string)
-	for _, configOption := range input {
-		splitKeyValue := strings.SplitN(configOption, "=", 2)
-		if len(splitKeyValue) == 2 {
-			if splitKeyValue[1] == "" {
-				valueMap[splitKeyValue[0]] = nil
-			} else {
-				valueMap[splitKeyValue[0]] = &splitKeyValue[1]
-			}
-		} else {
-			return nil, errors.New("Input doesn't contain =")
+func ConvertStringMapToAny(input map[string]string) map[string]any {
+	output := make(map[string]any)
+	for key, value := range input {
+		if intVal, err := strconv.ParseInt(value, 10, 64); err == nil {
+			output[key] = intVal
+			continue
+		}
+
+		if boolVal, err := strconv.ParseBool(value); err == nil {
+			output[key] = boolVal
+			continue
 		}
+
+		output[key] = json.RawMessage(value)
 	}
-	return valueMap, nil
+	return output
 }
 
-func BytesToGiB(volumeSizeGiB int64) int64 {
-	return volumeSizeGiB / GiB
-}
+func ConvertObjectToJsonString(input any) (string, error) {
+	bytes, err := json.Marshal(input)
+	if err != nil {
+		return "", err
+	}
 
-func GiBToBytes(volumeSizeGiB int64) int64 {
-	return volumeSizeGiB * GiB
+	return string(bytes), nil
 }
 
-func StringToIntPointer(input string) (*int64, error) {
-	output, err := strconv.ParseInt(input, 10, 64)
+// ConvertJsonStringToObject converts a given input into an object. The input should be a formatted JSON string. The object must be passed in by reference
+func ConvertJsonStringToObject(input string, object any) error {
+	err := json.Unmarshal(json.RawMessage(input), object)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	return &output, nil
+	return nil
 }
 
-func StringToBoolPointer(input string) (*bool, error) {
-	output, err := strconv.ParseBool(input)
+// ConvertObjectToObject converts a given object into another object. The to object must be passed in by reference
+func ConvertObjectToObject(from any, to any) error {
```

Comment:
Can you change the name and/or add a little more information to the comment. I'm a little confused about why this is needed and what it's doing

Reply:
I can add a little more to the description but it essential converts object types. In our case we want to change a map to an FSx struct object

Reply:
+1, In general these functions should make it more clear that they're just converting object types